From 38d899e0f288a24173a6c4c5898bb176102ca9b9 Mon Sep 17 00:00:00 2001
From: "Gary D. Gregory" <ggregory@apache.org>
Date: Mon, 23 Jul 2018 15:59:37 +0000
Subject: [PATCH] [VFS-669]
 org.apache.commons.vfs2.util.CombinedResources.loadResources(String) should
 not throw an NPE for the system class loader is null.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/vfs/trunk@1836503 13f79535-47bb-0310-9956-ffa450edef68
---
 .../commons/vfs2/util/CombinedResources.java  | 24 ++++++++++---------
 src/changes/changes.xml                       |  3 +++
 2 files changed, 16 insertions(+), 11 deletions(-)

diff --git a/commons-vfs2/src/main/java/org/apache/commons/vfs2/util/CombinedResources.java b/commons-vfs2/src/main/java/org/apache/commons/vfs2/util/CombinedResources.java
index e698d1ae20..e65b01763b 100644
--- a/commons-vfs2/src/main/java/org/apache/commons/vfs2/util/CombinedResources.java
+++ b/commons-vfs2/src/main/java/org/apache/commons/vfs2/util/CombinedResources.java
@@ -74,19 +74,21 @@ protected void loadResources(String resourceName) {
         if (loader == null) {
             loader = ClassLoader.getSystemClassLoader();
         }
-        resourceName = resourceName.replace('.', '/') + ".properties";
-        try {
-            final Enumeration<URL> resources = loader.getResources(resourceName);
-            while (resources.hasMoreElements()) {
-                final URL resource = resources.nextElement();
-                try {
-                    properties.load(resource.openConnection().getInputStream());
-                } catch (final IOException ignored) {
-                    // Ignore
+        if (loader != null) {
+            resourceName = resourceName.replace('.', '/') + ".properties";
+            try {
+                final Enumeration<URL> resources = loader.getResources(resourceName);
+                while (resources.hasMoreElements()) {
+                    final URL resource = resources.nextElement();
+                    try {
+                        properties.load(resource.openConnection().getInputStream());
+                    } catch (final IOException ignored) {
+                        // Ignore
+                    }
                 }
+            } catch (final IOException ignored) {
+                // Ignore
             }
-        } catch (final IOException ignored) {
-            // Ignore
         }
     }
 
diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index cac2865d3b..88fced6ffd 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -83,6 +83,9 @@ The <action> type attribute can be add,update,fix,remove.
       <action issue="VFS-668" dev="ggregory" type="update">
         Throw a NPE with a better message when a class loader is null.
       </action>
+      <action issue="VFS-669" dev="ggregory" type="fix">
+        org.apache.commons.vfs2.util.CombinedResources.loadResources(String) should not throw an NPE for the system class loader is null.
+      </action>
     </release>
     <release version="2.2" date="2017-10-06" description="New features and bug fix release.">
       <action issue="VFS-642" dev="pschumacher" type="update" due-to="ilangoldfeld">
